docs: update API and authentication pages (#7271)
langflow-ai/langflow: docs/docs/Configuration/configuration-authentication.md

:::

To generate a `LANGFLOW_SECRET_KEY`, follow these steps:

1. Run the command to generate and copy a secret to the clipboard.

<Tabs>
<TabItem value="unix" label="macOS/Linux">

```bash
# Copy to clipboard (macOS)
python3 -c "from secrets import token_urlsafe; print(f'LANGFLOW_SECRET_KEY={token_urlsafe(32)}')" | pbcopy

# Copy to clipboard (Linux)
python3 -c "from secrets import token_urlsafe; print(f'LANGFLOW_SECRET_KEY={token_urlsafe(32)}')" | xclip -selection clipboard

# Or just print
python3 -c "from secrets import token_urlsafe; print(f'LANGFLOW_SECRET_KEY={token_urlsafe(32)}')"
```
</TabItem>

<TabItem value="windows" label="Windows">

```bash
# Copy to clipboard
python -c "from secrets import token_urlsafe; print(f'LANGFLOW_SECRET_KEY={token_urlsafe(32)}')" | clip

# Or just print
python -c "from secrets import token_urlsafe; print(f'LANGFLOW_SECRET_KEY={token_urlsafe(32)}')"
```

</TabItem>
</Tabs>

2. Paste the value into your `.env` file:
```text
LANGFLOW_SECRET_KEY=dBuuuB_FHLvU8T9eUNlxQF9ppqRxwWpXXQ42kM2_fb
```

### LANGFLOW_NEW_USER_IS_ACTIVE

> When set to `True`, new users are automatically activated and can log in without requiring explicit activation by the superuser from the **Admin page**.

Reviewer: ```suggestion
When this option is set to `True`, new users are automatically activated and can log in without requiring explicit activation by the superuser from the **Admin page**.
```

By default, this variable is set to `False`.

```text
LANGFLOW_NEW_USER_IS_ACTIVE=False
```

## Start a secure Langflow server with authentication

Start a secure Langflow server with authentication enabled and secret key encryption using the variables described in [Authentication configuration values](/configuration-authentication#authentication-configuration-values).

> Once logged in as a superuser, you will create a new user on your server.

Reviewer: ```suggestion
Once you are logged in as a superuser, create a new user on your server.
```

### Start the Langflow server

1. Create a `.env` file and populate it with values for a secure server.
This server creates a superuser account, requires users to log in before using Langflow, and encrypts secrets with `LANGFLOW_SECRET_KEY`, which is added in the next step.

> Create a `.env` file with this configuration:

Reviewer: ```suggestion
Create a `.env` file with the following configuration:
```

```text
LANGFLOW_AUTO_LOGIN=False
LANGFLOW_SUPERUSER=administrator
LANGFLOW_SUPERUSER_PASSWORD=securepassword
LANGFLOW_SECRET_KEY=your_generated_key
LANGFLOW_NEW_USER_IS_ACTIVE=False
```

2. Generate a secret key for encrypting sensitive data.

> Generate your secret key using one of these commands:

Reviewer: ```suggestion
Generate your secret key using one of the following commands:
```

</TabItem>
</Tabs>

3. Paste your `LANGFLOW_SECRET_KEY` into the `.env` file.

4. Start Langflow with the configuration from your `.env` file.

```text
uv run langflow run --env-file .env
```

5. Verify the server is running. The default location is `http://localhost:7860`.

> ### Manage users as administrator

Reviewer: ```suggestion
### Manage users as an administrator
```

> 1. To complete your first time login as superuser, go to `http://localhost:7860/login`.

Reviewer: ```suggestion
1. To complete your first-time login as a superuser, go to `http://localhost:7860/login`.
```

2. Log in with your superuser credentials:
* Username: Value of `LANGFLOW_SUPERUSER` (for example, `administrator`)
* Password: Value of `LANGFLOW_SUPERUSER_PASSWORD` (for example, `securepassword`)

:::info
The default values are `langflow` and `langflow`.
:::

3. To manage users on your server, navigate to the `/admin` page at `http://localhost:7860/admin`.
Click your user profile image, and then click **Admin Page**.

As a superuser, you can create users, set permissions, reset passwords, and delete accounts.

> 4. To create a user, click **New User** and complete the fields:

Reviewer: ```suggestion
4. To create a user, click **New User**, and then complete the following fields:
```

* Username
* Password and Confirm Password

> * Select **Active** and deselect **Superuser** for the new user.

Reviewer: What are the field names where these items are being selected?
